webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -833,8 +852,24 @@ trait PromiseGeneration_BuildNodeContext {
           nodeTargets  =  allGroups.getTarget(info).map(_._2).toList
           timeMerge    =  System.nanoTime
           mergedProps  <- MergeNodeProperties.forNode(info, nodeTargets, nodeParam.map{case(k,v) => (k, v)}.toMap).toBox
+          nodeContextBefore  <- systemVarService.getSystemVariables(info, allNodeInfos, nodeTargets, globalSystemVariables, globalAgentRun, globalComplianceMode: ComplianceMode)
+          // Not sure if I should InterpolationContext or create a "EngineInterpolationContext
+          contextEngine = InterpolationContext(
+              info
+            , policyServer
+            , globalPolicyMode
+            , nodeContextBefore
+            , nodeParam.map{case(k, g) => (k, g.value)}.toMap
+          )
           _            =  {timeNanoMergeProp = timeNanoMergeProp + System.nanoTime - timeMerge}
-          nodeInfo     =  info.modify(_.node.properties).setTo(mergedProps.map(_.prop))
+          propsCompiled <- ZIO.foreach(mergedProps) { p =>
+                            for {
+                              x <- parseJValue(p.prop.toJson, contextEngine)
+                            } yield {
+                              p.copy(prop = NodeProperty(p.prop.config.getString("name"), x.toString.toConfigValue, None, None))
```

Comment:
There is a problem here
value of mergedProps is
List(Global(Quoted("### Managed by Rudder, edit with care ###")))), NodePropertyHierarchy(NodeProperty({
    # String: 1
    "name" : "test",
    # hardcoded value
    "provider" : "inherited",
    # String: 1
    "value" : "test"

propsCompiled is
List(Global(Quoted("### Managed by Rudder, edit with care ###")))), NodePropertyHierarchy(NodeProperty({
    # hardcoded value
    "name" : "test",
    # hardcoded value
    "value" : "JObject(List(JField(name,JString(test)), JField(value,JString(test)), JField(provider,JString(inherited)), JField(inheritMode,JNothing)))"
}